kong/router/transform.lua

```diff
@@ -277,13 +277,40 @@ do
 end
 
 
-local function sni_val_transform(_, p)
-  if #p > 1 and byte(p, -1) == DOT then
-    -- last dot in FQDNs must not be used for routing
-    return p:sub(1, -2)
+local function sni_op_transform(sni)
+  local op = OP_EQUAL
+
+  if byte(sni) == ASTERISK then
+    -- postfix matching
+    op = OP_POSTFIX
+
+  elseif byte(sni, -1) == ASTERISK then
+    -- prefix matching
+    op = OP_PREFIX
   end
 
-  return p
+  return op
+end
+
+
+local function sni_val_transform(op, sni)
+  -- prefix matching
+  if op == OP_PREFIX then
+    sni = sni:sub(1, -2)
+
+  else
+    if #sni > 1 and byte(sni, -1) == DOT then
+      -- last dot in FQDNs must not be used for routing
+      sni = sni:sub(1, -2)
+    end
```

Comment:
I think that this block should be the first logic in the function.

Reply:
Because the last character can't be `DOT` or `ASTERISK` at the same time, so there is no need to check this when `op == OP_PREFIX`